### PR TITLE
feat: changing cycles to discount in subscriptions and plans

### PR DESCRIPTION
**Pull request by mateus-picoloto**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| **Issue**         | https://mundipagg.atlassian.net/browse/PAOPN-575
| **What?**         | Changing cycles to discount in subscriptions and plans.
| **Why?**          | To be able to apply discount in all charges from subscriptions and plans.
| **How?**          | Changing cycles to discount in subscriptions and plans.

### CODE REVIEW

**Review by mauriciohaygert — APPROVED** [review]

File: src/Recurrence/Aggregates/Plan.php [hunk 1]
```diff
@@ -35,6 +33,8 @@ final class Plan extends AbstractEntity implements RecurrenceEntityInterface, Pr
     private $items;
     private $trialPeriodDays;
 
+    private $applyProductsCycleToDiscount;
```
Comment (RafaMelazzo) [inline src/Recurrence/Aggregates/Plan.php:36]: $applyDiscountInAllProductCycles

File: src/Recurrence/Aggregates/Plan.php [hunk 2]
```diff
@@ -383,6 +383,22 @@ public function setTrialPeriodDays($trialPeriodDays)
         $this->trialPeriodDays = $trialPeriodDays;
     }
 
+    /**
+     * @return bool
+     */
+    public function getApplyProductsCycleToDiscount()
```
Comment (RafaMelazzo) [inline src/Recurrence/Aggregates/Plan.php:389]: getApplyDiscountInAllProductCycles

File: src/Recurrence/Aggregates/Plan.php [hunk 3]
```diff
@@ -383,6 +383,22 @@ public function setTrialPeriodDays($trialPeriodDays)
...
+    {
+        return boolval($this->applyProductsCycleToDiscount);
+    }
+
+    /**
+     * @param bool $applyProductsCycleToDiscount
+     */
+    public function setApplyProductsCycleToDiscount($applyProductsCycleToDiscount)
```
Comment (RafaMelazzo) [inline src/Recurrence/Aggregates/Plan.php:397]: setApplyDiscountInAllProductCycles

**Review by RafaMelazzo — CHANGES_REQUESTED** [review]

**Review by RafaMelazzo — DISMISSED** [review]

File: src/Recurrence/Aggregates/Repetition.php [hunk 4]
```diff
@@ -211,8 +209,11 @@ public function getAvailablesInterval()
      */
     public function checkRepetitionIsCompatible(Repetition $repetitionObject)
     {
-        if (($this->getInterval() == $repetitionObject->getInterval()) &&
-            ($this->getIntervalCount() == $repetitionObject->getIntervalCount())) {
+        if (
+            ($this->getInterval() === $repetitionObject->getInterval()) &&
```
Comment (fabiano-mallmann) [inline src/Recurrence/Aggregates/Repetition.php:213]: Os parenteses não fazem sentido existirem e adicionar os "agregadores" segundo as PERs.

Comment (mateus-picoloto) [reply src/Recurrence/Aggregates/Repetition.php]: Ajustado

**Review by RafaMelazzo — APPROVED** [review]

**Review by fabiano-mallmann — APPROVED** [review]

**Pull request merged**